src/routes/_surfaces/workspace/explore/TimeRangeSelector.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  import Button from "$lib/components/Button.svelte";
+  import CaretDownIcon from "$lib/components/icons/CaretDownIcon.svelte";
+  import Menu from "$lib/components/menu/Menu.svelte";
+  import MenuItem from "$lib/components/menu/MenuItem.svelte";
+  import FloatingElement from "$lib/components/tooltip/FloatingElement.svelte";
+  import { setExploreSelectedTimeRangeAndUpdate } from "$lib/redux-store/explore/explore-apis";
+  import { getMetricsExploreById } from "$lib/redux-store/explore/explore-readables";
+  import { store } from "$lib/redux-store/store-root";
+  import { onClickOutside } from "$lib/util/on-click-outside";
+  import { makeAvailableTimeRanges } from "$lib/util/time-ranges";
+  import { tick } from "svelte";
+  import { getTimeRangeNameForButton, prettyFormatTimeRange } from "./utils";
+
+  export let metricsDefId: string;
+
+  $: metricsLeaderboard = getMetricsExploreById(metricsDefId);
+
+  $: selectedTimeRange = $metricsLeaderboard.selectedTimeRange;
+  $: availableTimeRanges = makeAvailableTimeRanges(
+    $metricsLeaderboard?.timeRange
+  );
+
+  $: if (!selectedTimeRange) {
+    selectedTimeRange = availableTimeRanges[0];
+  }
+
+  let timeSelectorMenu;
```

Comment:
Feels like hide/show of a list feature might be needed in a lot of places. Should we generalise this in some way?

Reply:
yeah, I think that the way we're doing menus needs to be simplified. We've half-generalized it right now.